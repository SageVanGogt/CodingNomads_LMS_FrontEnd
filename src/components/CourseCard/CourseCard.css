.course_card {
  width: 250px;
  height: 200px;
  margin: 20px;
  padding: 10px;
  background-color: white;
  font-family: 'Open Sans', sans-serif;
  border-radius: 5px;
  -webkit-box-shadow: 0 5px 6px -4px rgba(0, 0, 0, 0.22), 0 10px 13px -4px rgba(0, 0, 0, 0.34), inset 0 30px 30px 0 rgba(255, 255, 255, 0.35), inset 0 -30px 30px 0 rgba(0, 0, 0, 0.1);
  box-shadow: 0 5px 6px -4px rgba(0, 0, 0, 0.22), 0 10px 13px -4px rgba(0, 0, 0, 0.34), inset 0 30px 30px 0 rgba(255, 255, 255, 0.35), inset 0 -30px 30px 0 rgba(0, 0, 0, 0.1); }

.course_top {
  display: flex;
  justify-content: space-between; }

.Course_edit-btn {
  position: relative;
  background-color: #4885a2;
  border-radius: 5px;
  height: 20px;
  width: 40px;
  margin: 10px;
  font-weight: 600;
  color: white;
  border: none;
  cursor: pointer; }

.Course_name {
  margin: 10px;
  font-weight: 800; }

.Course_desc {
  margin-left: 10px;
  text-align: left;
  font-size: .9rem; }

.course_card hr {
  width: 200px;
  color: white;
  margin-bottom: 15px;
  border: 0;
  height: 1px;
  background-image: linear-gradient(to right, rgba(0, 0, 0, 0), rgba(0, 0, 0, 0.75), rgba(0, 0, 0, 0)); }
