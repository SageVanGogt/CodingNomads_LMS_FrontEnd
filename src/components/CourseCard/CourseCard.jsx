import React, { Component } from 'react';
import './CourseCard.css';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { updateCurrentCourse } from '../../actions/currentCourse';
import { COURSE_EDIT } from '../../constants/routes';

export class CourseCard extends Component {
  constructor(props) {
    super(props);
  }

  updateCourse = async () => {
    this.props.updateCurrentCourse(this.props);
    this.props.history.push(COURSE_EDIT);
  }

  editBtn = () => {
    return <button className="Course_edit-btn" onClick={this.updateCourse}>Edit</button>;
  }

  viewBtn = () => {
    return <button className="Course_edit-btn">View</button>;
  }

  render() {
    const { user, name, description } = this.props;

    const adminEditBtn = this.editBtn();
    const courseViewBtn = this.viewBtn();

    return (
      <div className='course_card'>
        <h2>{name}</h2>
        {user.roleId === 1 ? adminEditBtn : courseViewBtn}
        <hr/>
        <p>{description}</p>
      </div>
    );
  }
}

CourseCard.propTypes = {
  id: PropTypes.number,
  name: PropTypes.string,
  description: PropTypes.string,
  user: PropTypes.object
};

export const mapStateToProps = (state) => ({
  user: state.user
});

export const mapDispatchToProps = dispatch => ({
  updateCurrentCourse: (currentCourse) => dispatch(updateCurrentCourse(currentCourse))
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(CourseCard));