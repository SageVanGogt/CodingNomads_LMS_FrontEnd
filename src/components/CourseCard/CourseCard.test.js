import React from 'react';
import { shallow } from 'enzyme';
import { CourseCard, mapStateToProps } from './CourseCard';

describe('CourseCard', () => {
  let wrapper;
  let mockStudent;

  beforeEach(() => {
    mockStudent = { id: 1, roleId: 2 };
    wrapper = shallow(<CourseCard user={mockStudent} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot for admin permissions', () => {
    let mockAdmin = {
      id: 1,
      roleId: 1
    };
    wrapper = shallow(<CourseCard user={mockAdmin} />);

    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should return a state of the user', () => {
      let mockState = {
        tasks: [{}, {}],
        user: { id: 1, roleId: 1 }
      };
      let mappedProps = mapStateToProps(mockState);
      let actual = mappedProps.user;
      let expected = { id: 1, roleId: 1 };

      expect(actual).toEqual(expected);
    });
  });
});