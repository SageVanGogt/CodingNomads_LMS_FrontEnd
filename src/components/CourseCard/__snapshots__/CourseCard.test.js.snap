// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseCard should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CourseCard
    user={
        Object {
            "id": 1,
            "roleId": 2,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2 />,
        <button
          className="Course_edit-btn"
>
          View
</button>,
        <hr />,
        <p />,
      ],
      "className": "course_card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "View",
          "className": "Course_edit-btn",
        },
        "ref": null,
        "rendered": "View",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2 />,
          <button
            className="Course_edit-btn"
>
            View
</button>,
          <hr />,
          <p />,
        ],
        "className": "course_card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "View",
            "className": "Course_edit-btn",
          },
          "ref": null,
          "rendered": "View",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`CourseCard should match the snapshot for admin permissions 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CourseCard
    user={
        Object {
            "id": 1,
            "roleId": 1,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2 />,
        <button
          className="Course_edit-btn"
>
          Edit
</button>,
        <hr />,
        <p />,
      ],
      "className": "course_card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Edit",
          "className": "Course_edit-btn",
        },
        "ref": null,
        "rendered": "Edit",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2 />,
          <button
            className="Course_edit-btn"
>
            Edit
</button>,
          <hr />,
          <p />,
        ],
        "className": "course_card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Edit",
            "className": "Course_edit-btn",
          },
          "ref": null,
          "rendered": "Edit",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
